File: SuperheroApp/CharacterDetail/CharacterDetailContract.swift
//
//  CharacterDetailContract.swift
//  SuperheroApp
//
//  Created by Balazs Varga on 2018. 02. 12..
//  Copyright © 2018. W.UP. All rights reserved.
//

protocol CharacterDetailMvpView: MvpView {
    
    func showCharacter(character: Character)
}

protocol CharacterDetailMvpPresenter: MvpPresenter {
    
    func takeView(view: CharacterDetailMvpView)
    
    func loadCharacter(characterId: Int)
}

File: common/src/main/kotlin/digital/wup/superheroapp/common/charaterdetail/CharacterDetailContract.kt
package digital.wup.superheroapp.charaterdetail

import digital.wup.superheroapp.characters.domain.model.Character
import digital.wup.superheroapp.mvp.MvpPresenter
import digital.wup.superheroapp.mvp.MvpView

interface CharacterDetailMvpPresenter : MvpPresenter {
    fun takeView(view: CharacterDetailMvpView)

    fun dropView()

    fun loadCharacter(characterId: Int)
}

interface CharacterDetailMvpView : MvpView {

    fun showCharacter(character: Character)

    fun showNoCharacter()

    fun showErrorMessage(message: String)
}